daemon/mgr/container_storage.go

```diff
@@ -471,96 +471,60 @@ func (mgr *ContainerManager) setMountTab(ctx context.Context, c *Container) erro
 	return nil
 }
 
-func (mgr *ContainerManager) setRootfsQuota(ctx context.Context, c *Container) error {
-	logrus.Debugf("start to set rootfs quota, dir(%s)", c.MountFS)
+func (mgr *ContainerManager) setDiskQuota(ctx context.Context, c *Container, mounted bool) error {
+	var globalQuotaID uint32
 
-	if c.MountFS == "" {
-		return nil
-	}
-
-	rootfsQuota := quota.GetDefaultQuota(c.Config.DiskQuota)
-	if rootfsQuota == "" {
-		return nil
-	}
-
-	qid := "0"
-	if c.Config.QuotaID != "" {
-		qid = c.Config.QuotaID
-	}
-
-	id, err := strconv.Atoi(qid)
-	if err != nil {
-		return errors.Wrapf(err, "failed to change quota id(%s) from string to int", qid)
-	}
-
-	// set rootfs quota
-	_, err = quota.SetRootfsDiskQuota(c.MountFS, rootfsQuota, uint32(id))
-	if err != nil {
-		return errors.Wrapf(err, "failed to set rootfs quota, mountfs(%s), quota(%s), quota id(%d)",
-			c.MountFS, rootfsQuota, id)
-	}
-
-	return nil
-}
-
-func (mgr *ContainerManager) setMountPointDiskQuota(ctx context.Context, c *Container) error {
-	if c.Config.DiskQuota == nil {
-		if c.Config.QuotaID != "" && c.Config.QuotaID != "0" {
-			return fmt.Errorf("invalid argument, set quota-id without disk-quota")
-		}
-		return nil
-	}
-
-	var (
-		qid        uint32
-		setQuotaID bool
-	)
-
-	if c.Config.QuotaID != "" {
+	if c.Config.QuotaID != "" && c.Config.QuotaID != "0" {
 		id, err := strconv.Atoi(c.Config.QuotaID)
 		if err != nil {
 			return errors.Wrapf(err, "invalid argument, QuotaID(%s)", c.Config.QuotaID)
 		}
 
 		// if QuotaID is < 0, it means pouchd alloc a unique quota id.
 		if id < 0 {
-			qid, err = quota.GetNextQuotaID()
+			globalQuotaID, err = quota.GetNextQuotaID()
 			if err != nil {
 				return errors.Wrap(err, "failed to get next quota id")
 			}
 
 			// update QuotaID
-			c.Config.QuotaID = strconv.Itoa(int(qid))
+			c.Config.QuotaID = strconv.Itoa(int(globalQuotaID))
 		} else {
-			qid = uint32(id)
+			globalQuotaID = uint32(id)
 		}
 	}
 
-	if qid > 0 {
-		setQuotaID = true
-	}
-
-	// get rootfs quota
+	// get default quota
 	quotas := c.Config.DiskQuota
 	defaultQuota := quota.GetDefaultQuota(quotas)
-	if setQuotaID && defaultQuota == "" {
-		return fmt.Errorf("set quota id but have no set default quota size")
-	}
 
-	// parse diskquota regexe
+	// parse disk quota regexp
 	var res []*quota.RegExp
-	for path, size := range quotas {
-		re := regexp.MustCompile(path)
-		res = append(res, &quota.RegExp{Pattern: re, Path: path, Size: size})
+	for key, size := range quotas {
+		var err error
+		id := globalQuotaID
+		if id == 0 {
+			id, err = quota.GetNextQuotaID()
+			if err != nil {
+				return errors.Wrap(err, "failed to get next quota id")
+			}
+		}
+		paths := strings.Split(key, "&")
+		for _, path := range paths {
+			re := regexp.MustCompile(path)
```

Comment:
should we need to verify the path, like what `validateDiskQuota` does?

Reply:
we have check it when call validateDiskQuota.